Adding French translation

xerta555 opened the pull request:
French translation files can be merged.

xerta555 commented:
> Thanks a lot for working on this! Sorry about not replying promptly, but I was in the middle of updating our localization backend. That'll probably entail some minor syntactic changes to the .ftl files, but I'll handle that before merging this.

@jimporter Hello, np about your reply delay.

About the syntactc changes to the `.flt` files, tell me here when files will be ready i will update the French ones.

> In addition, just so you know, we're going to be putting this repository on ice for a bit while we focus on the iOS version. If you'd like to continue helping with localizing, feel free to take a look at that repo; I don't think it's quite ready for localizing yet, but it shouldn't be too long until it's ready.

About the iOS version, i will checking this repo sometimes and i will create the request accordingly.

As for the Android version, maybe i'm wrong, but i haven't seen a repository for this Lockbox version, maybe it's been added in the meantime?

linuxwolf commented:
Hello @xerta555 ,

The updates to Fluent are merged into our `master` branch now; you can check them out when you merge/rebase `master` into your branch.  The most recent commit of our en-US `.ftl` files should make things clear, and reach out if things are not clear.

The iOS app is starting with a focus on English for its initial pilot releases, but translations will be most welcome -- just know the application is still evolving.

The Anroid app has not started yet, but keep a watch on the `mozilla-lockbox` Github org for it!

And, again, thanks very much for this contribution!

xerta555 commented:
> The updates to Fluent are merged into our master branch now; you can check them out when you merge/rebase master into your branch. The most recent commit of our en-US .ftl files should make things clear, and reach out if things are not clear.

Hello, no problem about my contribution.
I came across this repo while running away on bugzilla for the WebExtensions API bugreports list, i won't comment on the change of direction taken by Mozilla a few months ago on this subject although i hope the APIs will be developed as soon as possible for Firefox. That's why i thought that participating in support at least at the languages translations level on this future extension would be a good thing.

Concerning the translation for the office version, i have just update the translations for the same files as before.

As for the iOS version, i will deal with it tomorrow if i have time or this weekend quietly. Compared to that, the decision to first develop the version for iPhones and then the version for Android seemed contrary to Mozilla's deployment habits.

I got a little scattered in my first paragraph about WebExtension support in Firefox because i intend to update as soon as possible some modules that became obsolete for the Android version of Firefox by reading the WebExtension compatibility policy and seeing that there is still no support for WebExtensions for this branch, so i decided to pause my development to focus on other projects more related to Android.

Friendly.

xerta555 commented:
Hello,
I just browse all folders in the lockbox-iOS version but i haven't see a revelant folder wich contain the US/EN strings sentences, the only one that i have found it this file: [https://github.com/mozilla-lockbox/lockbox-ios/blob/master/lockbox-ios/Common/Resources/Strings/en.lproj/Localizable.strings](https://github.com/mozilla-lockbox/lockbox-ios/blob/master/lockbox-ios/Common/Resources/Strings/en.lproj/Localizable.strings), but there is only 4 strings so i have some doubts..

If u can link me the proper folder i could make something.

linuxwolf commented:
Hello @xerta555 we've added mozilla-lockbox/lockbox-ios#440 to track on initial localization support.  We'd appreciate future discussion on the iOS app to happen over on that repo, (for now) on that issue if possible.

Thanks!

xerta555 commented:
> Hello @xerta555 we've added mozilla-lockbox/lockbox-ios#440 to track on initial localization support. We'd appreciate future discussion on the iOS app to happen over on that repo, (for now) on that issue if possible.

Hello, i have translate directly all indicated files referenced in your link.
I am not a pro in iOS, but in the first file, i don't understand the meaning of the sentences after `, comment:`.

Maybe it wasn't necessary to translate these sentences, in this case simple copy and paste between the initial/official repo and my changes will suffice, or maybe i did things well.

As i don't see where my commit have been referenced, you could find them in commits list of my fork: [here](https://github.com/xerta555/lockbox-ios/commits/master).

jimporter commented on src/webextension/locales/fr_FR/list.ftl:
```diff
@@ -0,0 +1,134 @@
+item-summary-new-title = Nouvel enregistrement
+item-summary-title =
+  { $length ->
+     [0]     (no site name)
```
The `(no site name)` string should probably be translated here.

xerta555 replied:
Fixed

jimporter commented on src/webextension/locales/fr_FR/list.ftl:
```diff
@@ -0,0 +1,134 @@
+item-summary-new-username = (entrer vos informations de connexion)
+item-summary-username =
+  { $length ->
+     [0]     (no username)
```
As above, the `(no username)` string should be translated.

xerta555 replied:
Fixed

jimporter commented on src/webextension/locales/fr_FR/list.ftl:
```diff
@@ -0,0 +1,134 @@
+breadcrumbs-item-new = Nouvel enregistrement
+breadcrumbs-item =
+  { $length ->
+     [0]     (no site name)
```
Also as above, the `(no site name)` string should be translated.

xerta555 replied:
Fixed

jimporter commented on src/webextension/locales/fr_FR/list.ftl:
```diff
@@ -0,0 +1,134 @@
+all-items-no-results = Aucun résultats
+
+item-fields-title = Nom
+item-fields-title-input
```
You'll need to add ` =` at the end of this line for it to work with the new Fluent syntax (like you did with `document` above).

xerta555 replied:
Fixed

jimporter commented on src/webextension/locales/fr_FR/list.ftl:
```diff
@@ -0,0 +1,134 @@
+item-fields-title-input
+  .placeholder = p. ex. banque primaire
+item-fields-origin = Adresse du site web
+item-fields-origin-input
```
Add ` =` to the end of this line too.

xerta555 replied:
Fixed

jimporter commented on src/webextension/locales/fr_FR/list.ftl:
```diff
@@ -0,0 +1,134 @@
+item-fields-origin-input
+  .placeholder = www.exemple.fr
+item-fields-username = Nom d'utilisateur
+item-fields-username-input
```
Add ` =` to the end of this line too.

xerta555 replied:
Fixed

jimporter commented on src/webextension/locales/fr_FR/list.ftl:
```diff
@@ -0,0 +1,134 @@
+item-fields-username = Nom d'utilisateur
+item-fields-username-input
+  .placeholder = nom@exemple.fr
+item-fields-copy-username
```
Add ` =` to the end of this line too.

xerta555 replied:
Fixed

jimporter commented on src/webextension/locales/fr_FR/list.ftl:
```diff
@@ -0,0 +1,134 @@
+item-fields-copy-username
+  .title = Copier le nom d'utilisateur dans le presse-papier
+item-fields-password = Mot de passe
+item-fields-copy-password
```
Add ` =` to the end of this line too.

xerta555 replied:
Fixed

jimporter commented on src/webextension/locales/fr_FR/list.ftl:
```diff
@@ -0,0 +1,134 @@
+
+homepage-linkaccount-title = Ajoutez une sécurité et une commodité sérieuses.
+homepage-linkaccount-description =
+    Créez maintenant un compte Firefox - ou ajoutez { product-title } à un compte
```
Tiny issue: you used a hyphen (`-`) here, but a dash (`–`) in the next line. I think you want a dash for both. (I'm very bad at French, so I could be wrong).

xerta555 replied:
Fixed

jimporter commented on src/webextension/locales/fr_FR/list.ftl:
```diff
@@ -0,0 +1,134 @@
+  .confirmLabel = Ignorer les changements
+  .cancelLabel = Revenir en arrière
+
+modal-delete = Supprimer cette entrée ?
```
Should there be a space before the question mark here? I vaguely recall that `?` and `!` have spaces before them in French, but you don't do this elsewhere. If all this is correct, feel free to ignore me; like I mentioned above, I'm not very good at French! :)

xerta555 replied:
About symbols `!` and `?` in French, the spaces in front of these two symbols are not considered as a French fault. This connotation is more appreciated, and discouraged, disliked by others because in Literature there are no spaces between the last word of a sentence and the exclamation or question mark.
The space is not prohibited or considered fundamentally as a fault in itself, but it is still appropriate to adapt according to the type of format of use that follows.

jimporter commented on src/webextension/locales/fr_FR/widgets.ftl:
```diff
@@ -0,0 +1,20 @@
+copy-to-clipboard-button = Copie
+copy-to-clipboard-copied = ✔ Copié
+
+filter-input-clear
```
Add = to the end of this line too. (I just noticed that I forgot to do this for this line in the `en-US` translation too, so I'll fix that version.)

xerta555 replied:
Fixed

jimporter commented on src/webextension/locales/fr_FR/widgets.ftl:
```diff
@@ -0,0 +1,20 @@
+filter-input-clear
+  .title = Éffacer
+
+modal-root
```
Add = to the end of this line too.

xerta555 replied:
Fixed

jimporter commented on src/webextension/locales/fr_FR/widgets.ftl:
```diff
@@ -0,0 +1,20 @@
+modal-root
+  .contentLabel = Dialogue modal
+
+password-input-show
```
Add = to the end of this line too.

xerta555 replied:
Fixed

jimporter commented on src/webextension/locales/fr_FR/widgets.ftl:
```diff
@@ -0,0 +1,20 @@
+
+password-input-show
+  .title = Afficher
+password-input-hide 
```
Add = to the end of this line too.

xerta555 replied:
Fixed

jimporter commented on src/webextension/locales/fr_FR/widgets.ftl:
```diff
@@ -0,0 +1,20 @@
+password-input-hide 
+  .title = Masquer
+
+panel-back-button
```
Add = to the end of this line too.

xerta555 replied:
Fixed

jimporter commented on src/webextension/locales/locales.json:
```diff
@@ -1 +1,2 @@
 ["en-US"]
+["fr-FR"]
```
This should be on the previous line, like so: `["en-US", "fr-FR"]`

xerta555 commented:
> This looks good, except for two things. First, your translation should go in src/webextension/locales/fr-FR (note the hyphen instead of the underscore). Second, see below.

It's now fixed, i have moved all translated files from `fr_FR` to `fr-FR` folder, and edited the `locales.json` file as u indicated.

devinreams commented:
Thank you @xerta555!🥇

I will take a look and test this out shortly...
